csm_web/scheduler/admin.py

```diff
@@ -49,273 +88,311 @@ def has_delete_permission(self, request, obj=None):
         return request.user.is_superuser
 
 
+@admin.register(Student)
+class StudentAdmin(CoordAdmin):
+    list_filter = ("active",)
+    list_display = ("name", "get_email", "get_course", "get_mentor", "section")
+    search_fields = ("user__email", "user__first_name", "user__last_name")
+    actions = ("drop_students", "undrop_students")
+    # autocomplete_fields = ("section", "user")
+
+    def get_fields(self, request, obj):
+        fields = ["name", "get_email", "get_course", "section", "get_attendances"]
+        fields.insert(4, "user" if request.user.is_superuser else "get_user")
+        return tuple(fields)
+
+    def get_readonly_fields(self, request, obj):
+        fields = ["get_course", "get_email", "name", "get_attendances"]
+        if not request.user.is_superuser:
+            fields.insert(0, "get_user")
+        return tuple(fields)
+
+    def has_delete_permission(self, request, obj=None):
+        return (
+            False
+        )  # delete sections + mentor by deleting sections, drop by deactivating
+
+    def formfield_for_foreignkey(self, db_field, request, **kwargs):
+        if not request.user.is_superuser:
+            courses = get_visible_courses(request.user)
+            if db_field.name == "section":
+                kwargs["queryset"] = Section.objects.filter(course__in=courses)
+        return super().formfield_for_foreignkey(db_field, request, **kwargs)
+
+    def get_queryset(self, request):
+        qs = super().get_queryset(request)
+        if request.user.is_superuser:
+            return qs
+        return qs.filter(section__course__in=get_visible_courses(request.user))
+
+    def drop_students(self, request, queryset):
+        queryset.update(active=False)
+
+    drop_students.short_description = (
+        "Drop student(s) from a section. This action is reversible, but it is still strongly "
+        "recommended that you double check the name/email/section/course first."
+    )
+
+    def undrop_students(self, request, queryset):
+        queryset.update(active=True)
+
+    undrop_students.short_description = "Re-enroll student(s) in the section they were previously dropped from."
+
+    # Custom fields
+
+    def get_email(self, obj):
+        return obj.user.email
+
+    get_email.short_description = "Email"
+
+    def get_user(self, obj):
+        return get_admin_link_for(obj.user, "admin:scheduler_user_change")
+
+    get_user.short_description = "User"
+
+    # TODO handle nullable section
+    def get_course(self, obj):
+        return get_admin_link_for(obj.section.course, "admin:scheduler_course_change")
+
+    get_course.short_description = "Course"
+
+    def get_mentor(self, obj):
+        return get_admin_link_for(obj.section.mentor, "admin:scheduler_mentor_change")
+
+    def get_attendances(self, obj):
+        attendance_links = []
+        for attendance in obj.attendance_set.all():
+            admin_url = reverse(
```

Comment:
Let's reuse `get_admin_link_for` instead of repeating ourselves.
Double check me, but I believe this would be:
```python
def get_attendances(self, obj):
    attendance_links = sorted(get_admin_link_for(attendance, 'admin:scheduler_attendance_change', f"Date {attendance.date}: {attendance.presence or '--'}") for attendance in obj.attendance_set.all())
    return format_html("".join(attendance_links))
```
Note the lack of list brackets inside `attendance_links` is deliberate, that's a generator comprehension if you haven't seen that before.